ui/src/pages/report/InputDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import Dialog from "@material-ui/core/Dialog";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogActions from "@material-ui/core/DialogActions";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import FormControl from "@material-ui/core/FormControl";
+import MenuItem from "@material-ui/core/MenuItem";
+
+export interface Selection {
+  items : string[]
+}
+
+export interface Field {
+  name : string
+  label : string
+  type : string | Selection
+}
+
+export interface InputDialogProps {
+  open : boolean
+  title : string
+  fields : Field[]
+  onClose : (state : any) => Promise<void>
+}
+
+export const InputDialog = (props : InputDialogProps) => {
+  const [ state, setState ] = useState<any>({});
```

Comment:
I wish that we didn't need this `any` and use a type parameter for the type of the choice. There was a bug in the original Dialog code that could have been avoided if we did that. Fine the way that it is.

Reply:
yeah, but I saw you actually improved this in another PR. will look at that shortly